pkg/scheduler/schedule_one.go

```diff
@@ -309,12 +309,21 @@ func (sched *Scheduler) bindingCycle(
 		return status
 	}
 
-	// Any failures after this point cannot lead to the Pod being considered unschedulable.
+	// Any failures after this point cannot lead to the Pod being considered unschedulable and so we create the PodScheduled condition.
 	// We define the Pod as "unschedulable" only when Pods are rejected at specific extension points, and PreBind is the last one in the scheduling/binding cycle.
 	//
 	// We can call Done() here because
 	// we can free the cluster events stored in the scheduling queue sonner, which is worth for busy clusters memory consumption wise.
 	sched.SchedulingQueue.Done(assumedPod.UID)
+	podScheduled := &v1.PodCondition{
+		Type:   v1.PodScheduled,
+		Status: v1.ConditionTrue,
+	}
+	newStatus := assumedPod.Status.DeepCopy()
+	podutil.UpdatePodCondition(newStatus, podScheduled)
+	if err := util.PatchPodStatus(ctx, sched.client, assumedPod, newStatus); err != nil {
+		logger.Error(err, "Error updating pod with PodScheduled condition", "pod", klog.KObj(assumedPod))
+	}
```

Comment:
Shouldn't the condition be set after binding?

Reply:
> `Any failures after this point cannot lead to the Pod being considered unschedulable` so I added it there.

Unschedulable != unscheduled in this context. This comment says that after this point, if the binding fails, pod can't be considered unschedulable, but if an error happens it is treated as scheduling error. In this case, pod is still unscheduled as it will need to go through scheduling cycle again. It means that status might be inaccurate, but fortunately will be updated to failure in `handleSchedulingFailure` ([ref](https://github.com/kubernetes/kubernetes/blob/a499facee693a1a83daadb82d88f7b51d324ffc5/pkg/scheduler/schedule_one.go#L1100-L1105)).